File: 2020-10-30-client-libraries/ruby/pagination.rb
require 'stripe'

Stripe.api_key = 'sk_test...'
puts 'Pagination in Ruby!'

# Cursor-based pagination
# customers = Stripe::Customer.list({limit: 10})
# customer_ids = []
# customers.each do |customer|
# 	customer_ids << customer.id
# end

# while customers.has_more do
# 	customers = Stripe::Customer.list({limit: 10, starting_after: customers.data.last.id})
# 	customers.each do |customer|
# 		customer_ids << customer.id
# 	end
# end
# puts customer_ids
# puts "# of customers: " + String(customer_ids.length)

# Auto-pagination
customer_ids = []
customers = Stripe::Customer.list({limit: 100})
customers.auto_paging_each do |customer|
	customer_ids << customer.id
end

puts customer_ids
puts "# of customers: " + String(customer_ids.length)

File: 2020-10-30-client-libraries/python/pagination.py
import stripe
import json

stripe.api_key = "sk_test..."
print ("Pagination in python!")

# Cursor-based pagination
# customer_ids = []
# customers = stripe.Customer.list(limit=10)
# for c in customers:
#   customer_ids.append(c.id)

# while customers.has_more:
#   customers = stripe.Customer.list(limit=10, starting_after=customers.data[-1].id)
#   for c in customers:
#     customer_ids.append(c.id)

# print(json.dumps(customer_ids,  encoding='ascii'))
# print ("# of customers: {:d}".format(len(customers)))

# Auto-pagination
customer_ids = []
customers = stripe.Customer.list(limit=100)
for c in customers.auto_paging_iter():
    customer_ids.append(c.id)

print(json.dumps(customer_ids,  encoding='ascii'))
print("# of customers: {:d}".format(len(customer_ids)))
